packages/block-editor/src/components/iframe/index.js

```diff
@@ -112,14 +112,19 @@ function Iframe( {
 	title = __( 'Editor canvas' ),
 	...props
 } ) {
-	const { resolvedAssets, isPreviewMode } = useSelect( ( select ) => {
-		const { getSettings } = select( blockEditorStore );
-		const settings = getSettings();
-		return {
-			resolvedAssets: settings.__unstableResolvedAssets,
-			isPreviewMode: settings.__unstableIsPreviewMode,
-		};
-	}, [] );
+	const { resolvedAssets, isPreviewMode, siteLocale, userLocale } = useSelect(
```

Comment:
Why do we need both siteLocale and userLocale? For me userLocale is just `isRTL` so we don't really need a new setting there.

Reply:
Good question.

The block editor should know about both site locale and user locale because:

1. it needs both the site lang and site RTL value to add the requisite attributes to the document (so that the content matches the frontend)
2. it should have the user locale available in case the editor canvas needs to render any editor-specific UI, such as placeholders.

On point 2, I could be wrong in my assumption, but my instinct tells me that the block placeholders are part of the editor UI, and therefore, the locales should be consistent.

 | Before | After |
|--------|--------|
| <img width="396" alt="Screenshot 2024-07-25 at 11 27 36 AM" src="https://github.com/user-attachments/assets/c8a87f04-6d41-4933-9f13-a774433a2041"> | <img width="378" alt="Screenshot 2024-07-25 at 11 27 45 AM" src="https://github.com/user-attachments/assets/61a9feab-2773-4e69-91d6-d09f5519ed5f"> | 

The iframe may not need to access userLocale here at all, but here I just wanted to provide an example of how to make the placeholder UI conform to the user locale (and not the site). See [above](https://github.com/WordPress/gutenberg/pull/63883#discussion_r1692110347).

I'm thinking a HOC placeholder in block-editor would be a better place to add the `lang` and `dir` attributes to the `<Placeholders />`, not the funky CSS I have in the latest commit of this PR. Anyway, that would be a follow up.

Reply:
No, my question was simpler than that :P. We already have `wp.i18n.isRTL` function to know whether the user local is rtl or not, we don't need a new setting for that.

Reply:
You know I love typing :trollface: 

I think I [remember realizing we just needed the site locale.](https://github.com/WordPress/gutenberg/pull/63883#discussion_r1689062491)

So, yeah, good point. "Locale" comprises language too, but `wp.i18n.getLocaleData()[""].lang` should get us the value for the lang attribute too should we need it. 👍🏻